### DISCUSSION TITLE
Should this work?

### DISCUSSION THREAD

**Discussion by nogginbox** [dotnet/maui#3769] [Q&A] [answered]
When working with a preview I often find myself asking the question; is it me, or the framework's fault? I ask this question a lot normally too, but it's more likely to be my fault then.

I'm trying to bind a Label to a property on a Behaviour that I've created. It works at load time and shows the initial text, but it blanks the label text when the behaviour property text changes. I'd like to know, have I don't the right thing, but there is a bug in MAUI Preview 10, or have I done the wrong thing?

This is the code for the property in ``MyBaviour``:
```
public string MyText
{
    get { return (string)GetValue(MyTextProperty); }
    set { SetValue(MyTextProperty, value); }
}

public static readonly BindableProperty MyTextProperty = BindableProperty.Create(nameof(MyText), typeof(string), typeof(MyBehaviour), default(string));
```

Here is the XAML that does the binding:
```
<Entry Text="{Binding InputText}">
    <Entry.Behaviors>
        <behavior:MyBehavior x:Name="TheBehavior" />
    </Entry.Behaviors>
</Entry>

<Label Text="{Binding MyText, Source={x:Reference TheBehavior}}" />
```

It really feels like this should work. The binding works first time (I've set some test default text in the behaviour constructor) and my understanding is that by using a ``BindableProperty`` it should update the XAML when MyText is set and it definitely should not stop showing any text.

**So, is this me, or a bug?**
If it's me I'd like to know what I'm doing wrong, if it's a bug, then I can create an issue if there is not one already.

**Comment by hartez** [comment, marked as answer]
Without seeing all the behavior code, it's hard to say for sure. But generally speaking, yes - that should work. I'd say you should file an issue.

**Reply by nogginbox** [reply]
Just managed to upgrade to preview 11 **and IT WORKS NOW!!**

**Reply by hartez** [reply]
Glad to hear it!
